test/specs/method_test.dart

```diff
@@ -67,6 +67,18 @@ void main() {
     );
   });
 
+  test('should create a lambda method if the value is cast', () {
+    expect(
+      Method((b) => b
+        ..name = 'returnsCastedValue'
+        ..returns = refer('Foo')
+        ..body = refer('bar').asA(refer('Foo')).code),
+      equalsDart(r'''
+        Foo returnsCastedValue() => (bar as Foo)
```

Comment:
should we be expecting a semicolon here or does that get added by something else?

Reply:
The semicolon gets added by something else.